### PR TITLE
Rename `number_of_*` functions

### PR DESCRIPTION
**Pull request by lgoettgens**
according to [this HackMD](https://hackmd.io/nRnyfrSxTVe5CzXidB1cBQ?both#number-of-%E2%80%A6) as discussed in the coding sprint.

cc @YueRen

### CODE REVIEW

File: docs/src/Combinatorics/graphs.md [hunk 1]
```diff
@@ -56,10 +56,10 @@ complete_bipartite_graph(n::Int64, m::Int64)
 edges(g::Graph{T}) where {T <: Union{Directed, Undirected}}
 has_edge(g::Graph{T}, source::Int64, target::Int64) where {T <: Union{Directed, Undirected}}
 has_vertex(g::Graph{T}, v::Int64) where {T <: Union{Directed, Undirected}}
+number_of_edges(g::Graph{T}) where {T <: Union{Directed, Undirected}}
+nv(g::Graph{T}) where {T <: Union{Directed, Undirected}}
```
Comment (fingolfin) [inline docs/src/Combinatorics/graphs.md]: Why is `ne` renamed to `number_of_edges` but `nv` (and elsewhere in this file: `nvertices`) left as it is, without a `number_of_vertices` being added?

Comment (lgoettgens) [reply docs/src/Combinatorics/graphs.md]: This is an (more or less) alphabetic list, `nv` just comes at the bottom

**Comment by lgoettgens** [comment]
This now contains all changes to `number_of_*` functions from the HackMD.
I would appreciate this getting merged in a timely manner, before too many conflicts arise.

Just as a side-note: Nemo has its own `number_of_partitions` with slightly different semantics regarding overflows. I deactivated the failing JuLie tests for the time being, as all of this will get unified and overhauled by https://github.com/oscar-system/Oscar.jl/pull/3159 (by @joschmitt) anyway.

**Comment by joschmitt** [comment]
> Just as a side-note: Nemo has its own `number_of_partitions` with slightly different semantics regarding overflows. I deactivated the failing JuLie tests for the time being, as all of this will get unified and overhauled by #3159 (by @joschmitt) anyway.

Oh, great. (And they are calling different C functions. So many choices.)

**Review by fingolfin — APPROVED** [review]

**Pull request merged**